lib/resty/rediscluster.lua

```diff
@@ -229,20 +229,15 @@ function _M.fetch_slots(self)
     local serv_list = self.config.serv_list
     local serv_list_cached = slot_cache[self.config.name .. "serv_list"]
 
-    local serv_list_combined
+    local serv_list_combined = serv_list or {}
```

Comment:
is this safe? because below we're injecting entries into `serv_list_combined`, which is effectively `self.config.serv_list`, so we are changing our original configuration?

Shouldn't this be a copy of that table?

Reply:
`serv_list_combined` = original configuration + cached IPs = `serv_list` + `serv_list_cached.serv_list`.

Reply:
not sure we're talking about the same thing here.

Tables are passed by reference, so `serv_list_combined` == `serv_list` == `self.config.serv_list`, they are all the same table.

Now we add to that table the entries in `serv_list_cached.serv_list`. 

The next time we call this function, `self.config.serv_list` already has the entries from the previous call. Now we're adding a new set of entries again. So the table will grow with each call to this function, since we keep adding.